### ISSUE TITLE
Add superpixel layer

### ISSUE THREAD

**Issue by constantinpape** [napari/napari#1270]
## 🚀 Feature

Add a new layer type, that supports selecting, merging and detaching smaller fragments (superpixels or supervoxels) into larger objects.

## Motivation

This kind of functionality is very helpful to proofread large segmentations that were produced by a superpixel + agglomeration approach. I think it can be useful in other contexts as well.

## Pitch

In the following, I will refer to superpixels/supervoxels as "fragments" and collections of supervoxels as "segments"

The layer needs (at least) the following functionality:
- Select individual fragments inside of fragments; highlight the selection. 
- Detach fragments from segments; merge fragments into existing segments.
- Initialize the assignment of fragments to segments (e.g. with result from some agglomeration algorithm).

This layer should probably inherit some functionality from the normal labels layer, e.g. we also want to be able to paint.

## Additional context

This kind of functionality is supported by the proofreading tools from the Saalfeld Lab and very convenient in the context of connectomics segmentations:

https://github.com/saalfeldlab/bigcat
https://github.com/saalfeldlab/paintera

@jni was involved in some of the developments.

**Comment by jni** [comment]
I forgot I campaigned for the term "fragments", thanks for reminding me! 😂

👍 👍 👍 👍

**Comment by constantinpape** [comment]
> I forgot I campaigned for the term "fragments", thanks for reminding me!

Yes, "fragments" and "segments" are also my favorite terms for this ;).

**Comment by VolkerH** [comment]
I feel this issue should be linked to this one https://github.com/napari/napari/issues/1235#issuecomment-631882278 even if only for illustration purposes.

edited to add:

If the functionality is not in the labels layer (as in the example by @jni in #1235) there will have to be some way to turn the merged superpixels into labels.
